### ISSUE TITLE
Add GitHub action to push latest sha release as new semver to internal PyPI

### ISSUE THREAD

**Issue by VersusFacit** [dbt-labs/dbt-redshift#724]
Using in part the release process from existing releases, write a GitHub action to send a new main sha to the internal PyPI repo. Draw from prior art of `release.yml`

Requires also adding a new `dbt-releases/push-to-internal-pypi.yml` workflow which is the final step of the parent action (a near duplicate of our existing release GHA architecture). This will point to the code artifacts server instead of the current public PyPI endpoint.

GHA: can't test locally so takes longer.
AWS auth: could result in issues

**Comment by VersusFacit** [comment]
MVP ready to go in my forked repo. [Successful build and upload job](https://github.com/VersusFacit/dbt-redshift/actions/runs/8198409167/job/22421883192) (no testing yet, [see Slack for discussion](https://dbt-labs.slack.com/archives/C06LW74PPNC/p1709869824129899) about testing internal releases ) 

![Image](https://github.com/dbt-labs/dbt-redshift/assets/67295367/ac7abf36-8021-437c-a8b3-4a3a12c70984)

The 1.8.0b1 was there. The 1.8.0b2 was automated.

Blocked my the following convos happening [in Slack](https://dbt-labs.slack.com/archives/C06LW74PPNC/p1709869824129899): 
1. Emily and I will discuss how to professionalize
2. Dash and I will harden the security if needed from here (may not be!)
3. Michelle and Chenyu will align with us on the final semver scheme for Mantle's needs, although I have a dummy script already written for doing a build number semver scheme
```
 1. Query AWS for latest release
 2. trim off its build number
 3. create release number <current version>+<last build number+1> 
 4. rename the wheel and tar gz with this release number <major>.<minor>.<release>b<beta number>+<build no.>
```

**Comment by martynydbt** [comment]
When do we expect this to be unblocked?

**Comment by VersusFacit** [comment]
This is really a duplicate ticket so I'm removing it.
